library/common/network/mobile_utility.cc

```diff
@@ -0,0 +1,42 @@
+#include "library/common/network/mobile_utility.h"
+
+#include "envoy/common/platform.h"
+
+#include "source/common/common/assert.h"
+
+#ifdef SUPPORTS_GETIFADDRS
+#include <ifaddrs.h>
+#endif
+
+namespace Envoy {
+namespace Network {
+
+std::vector<std::string> MobileUtility::enumerateInterfaces() {
```

Comment:
can we write a quick test?

Reply:
This isn't really very testable since it's mapping system output. Here's the extent of Envoy testing on this functionality: https://github.com/envoyproxy/envoy/blob/main/test/common/network/utility_test.cc#L228

Since this function doesn't return a pointer, there's not even a non-null assertion to be made here.

Reply:
If Envoy is expecting non null. Can we expect and non empty vector?

Reply:
Discussed offline, but an empty vector is potentially a valid result.